utils/fixed_decimal/src/decimal.rs

```diff
@@ -2188,165 +1984,139 @@ fn test_float() {
             input: 1.234567,
-            precision: DoublePrecision::SignificantDigits(2, RoundingMode::HalfExpand),
+            precision: DoublePrecision::SignificantDigits(2),
             expected: "1.2",
         },
         TestCase {
             input: 1.234567,
-            precision: DoublePrecision::SignificantDigits(4, RoundingMode::HalfExpand),
+            precision: DoublePrecision::SignificantDigits(4),
             expected: "1.235",
         },
         TestCase {
             input: 1.234567,
-            precision: DoublePrecision::SignificantDigits(10, RoundingMode::HalfExpand),
+            precision: DoublePrecision::SignificantDigits(10),
             expected: "1.234567000",
         },
         TestCase {
             input: 888999.,
-            precision: DoublePrecision::SignificantDigits(1, RoundingMode::HalfExpand),
+            precision: DoublePrecision::SignificantDigits(1),
             expected: "900000",
         },
         TestCase {
             input: 888999.,
-            precision: DoublePrecision::SignificantDigits(2, RoundingMode::HalfExpand),
+            precision: DoublePrecision::SignificantDigits(2),
             expected: "890000",
         },
         TestCase {
             input: 888999.,
-            precision: DoublePrecision::SignificantDigits(4, RoundingMode::HalfExpand),
+            precision: DoublePrecision::SignificantDigits(4),
             expected: "889000",
         },
         TestCase {
             input: 988999.,
-            precision: DoublePrecision::SignificantDigits(1, RoundingMode::HalfExpand),
+            precision: DoublePrecision::SignificantDigits(1),
             expected: "1000000",
         },
         TestCase {
             input: 99888.,
-            precision: DoublePrecision::SignificantDigits(1, RoundingMode::HalfExpand),
+            precision: DoublePrecision::SignificantDigits(1),
             expected: "100000",
         },
         TestCase {
             input: 99888.,
-            precision: DoublePrecision::SignificantDigits(2, RoundingMode::HalfExpand),
+            precision: DoublePrecision::SignificantDigits(2),
             expected: "100000",
         },
         TestCase {
             input: 99888.,
-            precision: DoublePrecision::SignificantDigits(3, RoundingMode::HalfExpand),
+            precision: DoublePrecision::SignificantDigits(3),
             expected: "99900",
         },
         TestCase {
             input: 9.9888,
-            precision: DoublePrecision::SignificantDigits(1, RoundingMode::HalfExpand),
+            precision: DoublePrecision::SignificantDigits(1),
             expected: "10",
         },
         TestCase {
             input: 9.9888,
-            precision: DoublePrecision::SignificantDigits(2, RoundingMode::HalfExpand),
-            expected: "10",
+            precision: DoublePrecision::SignificantDigits(2),
+            expected: "10.0", // TODO: Discussion, shall we make it "10".
```

Comment:
In this case, the number is 9.9888, when we specify the significant digits to be 2, this will result to half_even(-1) which will result 10.0. 

is this the right behaviour then ?

Reply:
We can do truncate after `half_even` to avoid this behaviour if we do not want it.

Reply:
Yes; significant digits rounding should drop a single trailing zero if the number increased in magnitude. Don't just call `truncate` or `strip` because we only want to remove one of the trailing zeros. For example: `9.999` with 3 significant digits should result in `10.0`

We may want to make this a helper function and/or have documentation in case people want to do this outside of the `try_from_f64` function.

Reply:
I have added a condition to check if the magnitude has been increased or not. if it is increased, I truncate by `position + 1`